api/services/registration_service.py

```diff
@@ -88,15 +88,16 @@ def update_registration(self, registration, jsonData):
         if not registration:
             return
 
-        if not (registration.program_id and self.has_default_program_selection):
-            registration.program_id = self.selected_program_id
-
         self.user_id = self.create_user(jsonData) if self.selected_program_id else None
 
         if not self.has_default_program_selection:
             registration.signup_date = datetime.now()
 
         if registration.status != models.Registration.RegistrationStatus.COMPLETE:
+            if not (registration.program_id and self.has_default_program_selection):
+                # Update the registration program id if the registration is not completed.
+                registration.program_id = self.selected_program_id
+
             registration.status = models.Registration.RegistrationStatus.INCOMPLETE
```

Comment:
This logic will only work if a new user program is created for user. How new user program will be created?

Reply:
We are getting the latest active {IN-PROGRESS} user program in user program detail variable. and if the user has program status other than in-progress then we are creating new user program with user id and if the user has program status in-progress then we are updating the user program with user program details.

Reply:
In this registration logic if the registration status of the user is not complete means pending or incomplete then we updating the program id in the registration table for the user else not.

Reply:
As per this logic, we are trying to get registration of the user and if found then we are updating the user_program
else creating. For existing user, this will always come to update function.

<img width="844" alt="image" src="https://user-images.githubusercontent.com/54376313/199493455-fd92234e-5e9f-42fd-ac19-f4959ccfde4d.png">

Reply:
Yes, as per this logic the existing user always come through update registration function. But for existing user if the registration status is completed and if the user one more time came across registration flow then the program_id should not be get updated.